### PR TITLE
Housekeeping

### LINKED ISSUE #2259: [RFE] Color code minimap based on team, not player
This is a feature request, not a bug.

When playing team games with large numbers of players (eg:8) it would be useful to have an option to configure the minimap so that each team gets a color (eg:red and blue) as opposed to each player's color appearing.

### PR DESCRIPTION
**Pull request by SJuliez**
Even youtubers sometimes don't seem to configure anything in MM which makes me think that the MM default config should be improved already. This and other things:

- Replaced the unit label border teamcolor sub-option with a general team coloring option that also applies to the minimap. Also changed the ally color to more of a blue. While player coloring may have applications, I believe teamcoloring has no downside as a default option and is therefore on by default. Resolves #2259 
- Removed the option from the Client Settings to not show a hex tooltip in hexes without units (i.e., never show the terrain). This prevents a player from seeing a bridge's or building CF, armor or any other complicated terrain with (I think) no remedy but to set this precise option in the Client Settings. Not useful on any but the simplest maps.
- Set the tooltip to show weapons, armor and the portrait by default. These are all very useful (or nice in the case of portraits) and players shouldn't have to first turn them on.
- Added the keybind to toggle the Keybinds Board Overlay directly into the header text to clarify that this can be easily toggled and made it more opaque to improve readability.
- The Keybinds Board Overlay will now show some relevant keybinds in the Board Editor
- Set the "Darken Map at night" option to default off. I noticed with a youtuber that this doesn't read well ("Oh this is a blue map set") and it's probably more annoying than helpful after a while (i.e., 10 seconds).
- Set the terrain shadows to default on. People don't see to have many issues with them and I do think they improve the overall look.
- Removed the option "Minimap can be shown" from the Client Settings. Instead the minimap will be shown and re-shown when it was on before and not be opened by MM again if it was closed by the player.
- Set "Mouse Wheel Zoom" to default true, as zooming with the wheel instead of scrolling is standard game behavior.
- Set "Flip mouse wheel" to default true as zooming in with wheel up is also standard game behavior.
- The minimap now respects the "Flip mouse wheel" setting.
- Added a keybind for toggling hex coords, Ctrl-G. Some games such as Civ use this shortcut to show a grid; we dont have a grid, but coords are the next best thing.
- Added a menu shortcut for the Client Settings, Alt-C. Should have done this a million clicks before.
- In the Client Settings, set the font of the advanced settings value (the text field) to the SANS_SERIF logical font which will make the unicode characters appear there correctly (at least on my system). 
- Made the Client Settings and Boardview use PreferenceChangeEvents to show changes directly instead of unnecessary boardview calls.
- Made the Client Settings actually revert changes when it's cancelled.

image image

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/GUIPreferences.java [hunk 1]
```diff
@@ -312,14 +312,20 @@ protected GUIPreferences() {
 
         store.setDefault(FOV_HIGHLIGHT_RINGS_RADII, "5 10 15 20 25");
         store.setDefault(FOV_HIGHLIGHT_RINGS_COLORS_HSB, "0.3 1.0 1.0 ; 0.45 1.0 1.0 ; 0.6 1.0 1.0 ; 0.75 1.0 1.0 ; 0.9 1.0 1.0 ; 1.05 1.0 1.0 ");
-
+        store.setDefault(FOV_HIGHLIGHT,false);
+        store.setDefault(FOV_HIGHLIGHT_ALPHA, 40);
+        store.setDefault(FOV_DARKEN,true);
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/GUIPreferences.java:317]:
```suggestion
        store.setDefault(FOV_DARKEN, true);
```

File: megamek/src/megamek/client/ui/swing/GUIPreferences.java [hunk 2]
```diff
@@ -312,14 +312,20 @@ protected GUIPreferences() {
 
         store.setDefault(FOV_HIGHLIGHT_RINGS_RADII, "5 10 15 20 25");
         store.setDefault(FOV_HIGHLIGHT_RINGS_COLORS_HSB, "0.3 1.0 1.0 ; 0.45 1.0 1.0 ; 0.6 1.0 1.0 ; 0.75 1.0 1.0 ; 0.9 1.0 1.0 ; 1.05 1.0 1.0 ");
-
+        store.setDefault(FOV_HIGHLIGHT,false);
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/GUIPreferences.java:315]:
```suggestion
        store.setDefault(FOV_HIGHLIGHT, false);
```

File: megamek/src/megamek/client/ui/swing/GUIPreferences.java [hunk 3]
```diff
@@ -312,14 +312,20 @@ protected GUIPreferences() {
...
+        store.setDefault(FOV_HIGHLIGHT_ALPHA, 40);
+        store.setDefault(FOV_DARKEN,true);
+        store.setDefault(FOV_DARKEN_ALPHA, 100);
+        store.setDefault(FOV_STRIPES, 35);
+        store.setDefault(FOV_GRAYSCALE, "false");
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/GUIPreferences.java:320]: I know this is old, but why is this saved as a string "false"? Any ideas?

Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/GUIPreferences.java:320]: Why were these moved?

File: megamek/src/megamek/client/ui/swing/boardview/BoardView1.java [hunk 4]
```diff
@@ -1050,6 +1050,28 @@ public void performAction() {
...
+        controller.registerCommandAction(KeyCommandBind.TOGGLE_HEX_COORDS.cmd,
+                new CommandAction() {
+
+                    @Override
+                    public boolean shouldPerformAction() {
+                        if (shouldIgnoreKeyCommands()) {
+                            return false;
+                        } else {
+                            return true;
+                        }
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/boardview/BoardView1.java:1064]:
```suggestion
                        return !shouldIgnoreKeyCommands();
```

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java [hunk 5]
```diff
@@ -303,6 +301,27 @@ public void mouseDragged(MouseEvent e) {
...
+    private boolean savedTeamColoring;
+    private boolean savedUnitLabelBorder;
+    private boolean savedShowDamageDecal;
+    private boolean savedShowDamageLabel;
+    private String savedFovHighlightRingsRadii;
+    private String savedFovHighlightRingsColors;
+    private int savedFovHighlightAlpha;
+    private int savedFovDarkenAlpha;
+    private int savedNumStripesSlider;
+    HashMap<String, String> savedAdvancedOpt = new HashMap<>();
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:324]: Before I continue I have to get clarification on this: Why are we saving values for a revert and not just having the update only occur when the dialog is saved?

**Review by Windchild292 — COMMENTED** [review]
One important clarification about CommonSettingsDialog is required before I can continue my review.

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java (hunk 5 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:324]: Because these settings immediately show on the board or minimap and to do this they must be saved to the GUIPrefs.

File: megamek/src/megamek/client/ui/swing/GUIPreferences.java (hunk 3 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/GUIPreferences.java:320]: Hmmmm no :)

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java (hunk 5 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:324]: That makes sense. I'll finish my review post-D&D

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java [hunk 6]
```diff
@@ -320,34 +339,13 @@ public CommonSettingsDialog(JFrame owner, ClientGUI cg) {
...
-        JScrollPane advancedSettingsPane = new JScrollPane(advancedSettingsPanel);
-        panTabs.add("Advanced", advancedSettingsPane);
+        super(owner, Messages.getString("CommonSettingsDialog.title"), true);
 
+        JTabbedPane panTabs = new JTabbedPane();
         setLayout(new BorderLayout());
         getContentPane().add(panTabs, BorderLayout.CENTER);
         getContentPane().add(getButtonsPanel(), BorderLayout.PAGE_END);
-
+
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:348]:
```suggestion

```

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java [hunk 7]
```diff
@@ -605,7 +594,7 @@ private JPanel getSettingsPanel() {
         row = new ArrayList<>();
         row.add(getFocus);
         comps.add(row);
-
+
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:597]:
```suggestion

```

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java [hunk 8]
```diff
@@ -1776,98 +1765,50 @@ private JPanel getButtonOrderPanel(){
...
+        targetingPhaseCommands = new DefaultListModel<StatusBarPhaseDisplay.PhaseCommand>();
+        phasePane.add("Targeting", getButtonOrderPane(targetingPhaseCommands,
+                TargetingPhaseDisplay.TargetingCommand.values()));
+        
+        return buttonOrderPanel;
+    }
+
+    /** Constructs the button ordering panel for one phase. */ 
+    private JScrollPane getButtonOrderPane(DefaultListModel<PhaseCommand> list, 
+            StatusBarPhaseDisplay.PhaseCommand commands[]) {
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:1798]: This is not a Java-style array.

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java [hunk 9]
```diff
@@ -1202,10 +1224,6 @@ public void itemStateChanged(ItemEvent event) {
             stampFormatLabel.setEnabled(stampFilenames.isSelected());
         } else if (source.equals(fovInsideEnabled)) {
             guip.setFovHighlight(fovInsideEnabled.isSelected());
-            if ((clientgui != null) && (clientgui.bv != null)) {
-                clientgui.bv.clearHexImageCache();
-                clientgui.bv.repaint();
-            }
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:1208]: Just checking, but we don't need to do either of these to properly update the board with the newly changed state?

**Review by Windchild292 — COMMENTED** [review]
Two more comments, but I think this looks good.

File: megamek/src/megamek/client/ui/swing/GUIPreferences.java (hunk 3 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/GUIPreferences.java:320]: It's actually treated as a boolean everywhere else, so this is just wrong. I'll correct it.

Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/GUIPreferences.java:320]: I moved a few of the lines around to group them up better. May not have been terribly important for these specific lines, I don't remember.

File: megamek/src/megamek/client/ui/swing/boardview/BoardView1.java (hunk 4 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/boardview/BoardView1.java:1064]: Oh indeed :) Never noticed. I'll correct this for all the other binds too.

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java (hunk 6 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:348]: Not sure what's wrong with this line. No tabs. A bit of breathing space for the code ;)

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java (hunk 7 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:597]: This empty line separates the option blocks. Not different from many before and after. Dont think this should be changed.

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java (hunk 9 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:1208]: No, these lines are actually nonsense I may have put in myself a few years ago. The boardview does preferencechange-listening and can update when such an event is fired. It just needs the right checks for what pref has changed to update accordingly. It could even update without checking what has changed but I think that's bad form, as all hexes have to be redrawn etc.

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java (hunk 8 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java]: PhaseCommand[] commands

**Comment by SJuliez** [comment]
OK, added in the suggestions as commented.

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java (hunk 7 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:597]: This removes the extra spacing from it, which would be automatically removed in a further commit. Just dealing with future blankspace changes now.

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java (hunk 6 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:348]: This removes the extra spacing from it, which would be automatically removed in a further commit. Just dealing with future blankspace changes now.

**Review by Windchild292 — APPROVED** [review]
